src/components/manage/Blocks/Table/Cell.jsx

```diff
@@ -1,188 +1,126 @@
-
-      const inlineToolbarPlugin = createInlineToolbarPlugin({
-        structure: this.draftConfig.richTextEditorInlineToolbarButtons,
-      });
-
-      this.state = {
-        editorState,
-        inlineToolbarPlugin,
-      };
-    }
-
-    this.onChange = this.onChange.bind(this);
+const CellComponent = (props) => {
+  const {
+    value,
+    draftJs,
+    draftJsInlineToolbarPlugin,
+    isTableBlockSelected,
+    onSelectCell,
+    row,
+    cell,
+    editable,
+    detached,
+    disableNewBlocks,
+    onSelectBlock,
+    onAddBlock,
+    index,
+  } = props;
+
+  const node = useRef();
+
+  const { EditorState, convertFromRaw } = draftJs;
+  const createInlineToolbarPlugin = draftJsInlineToolbarPlugin.default;
+  let draftConfig;
+
+  const [editorState, setEditorState] = useState();
+
+  const [inlineToolbarPlugin, setInlineToolbarPlugin] = useState();
+  if (!__SERVER__) {
+    draftConfig = config.settings.richtextEditorSettings(props);
+    let editorstate;
+    editorstate = EditorState.createWithContent(convertFromRaw(value));
+
+    const inlineToolbarPlugin = createInlineToolbarPlugin({
+      structure: draftConfig.richTextEditorInlineToolbarButtons,
+    });
+    setEditorState(editorstate);
+    setInlineToolbarPlugin(inlineToolbarPlugin);
   }
 
-  /**
-   * Component did mount lifecycle method
-   * @method componentDidMount
-   * @returns {undefined}
-   */
-  componentDidMount() {
-    if (this.node) {
-      const onFocus = this.node.editor._onFocus;
-      this.node.editor._onFocus = (event) => {
+  const previsTableBlockSelected = usePrevious(isTableBlockSelected);
+
+  useEffect(() => {
+    if (node) {
+      const onFocus = node.editor._onFocus;
+      node.editor._onFocus = (event) => {
         onFocus(event);
-        this.props.onSelectCell(this.props.row, this.props.cell);
+        onSelectCell(row, cell);
       };
     }
-  }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
 
-  /**
-   * Component will receive props
-   * @method componentWillReceiveProps
-   * @param {Object} nextProps Next properties
-   * @returns {undefined}
-   */
-  UNSAFE_componentWillReceiveProps(nextProps) {
+  useEffect(() => {
     if (
-      nextProps.isTableBlockSelected !== this.props.isTableBlockSelected &&
-      this.props.cell === 0 &&
-      this.props.row === 0
+      isTableBlockSelected !== previsTableBlockSelected &&
+      cell === 0 &&
+      row === 0
     ) {
-      this.node.focus();
+      node.focus();
     }
-  }
+  }, [cell, isTableBlockSelected, previsTableBlockSelected, row]);
 
-  /**
-   * Change handler
-   * @method onChange
-   * @param {object} editorState Editor state.
-   * @returns {undefined}
-   */
-  onChange(editorState) {
-    this.setState({ editorState }, () => {
-      this.props.onChange(this.props.row, this.props.cell, editorState);
-    });
-  }
+  const onChange = (editorState) => {
+    setEditorState(onChange(row, cell, editorState));
```

Comment:
@Tishasoumya-02 It's an unnecessary state reset here. The setEditorState doesn't accept a callback like setState. You need to explicitly listen for state change inside a useEffect.